### PR TITLE
[SW-1383] Fix cloud up in external backend manual mode

### PR DESCRIPTION
**Pull request by jakubhava**
I also did a bit of deprecations as the original name was way too long

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala [hunk 1]
```diff
@@ -296,6 +299,10 @@ class ExternalH2OBackend(val hc: H2OContext) extends SparklingBackend with Exter
       """)
     }
 
+    if (conf.clusterSize.isEmpty) {
```
Comment (jakubhava) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala]: Require num of external h2o nodes in both auto and manual mode

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala [hunk 2]
```diff
@@ -35,7 +36,10 @@ trait ExternalBackendConf extends SharedBackendConf {
   def h2oClusterHost = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1).map(_.split(":")(0))
   def h2oClusterPort = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1).map(_.split(":")(1).toInt)
 
-  def numOfExternalH2ONodes = sparkConf.getOption(PROP_EXTERNAL_H2O_NODES._1)
+  @DeprecatedMethod("clusterSize")
+  def numOfExternalH2ONodes = clusterSize
+
+  def clusterSize = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_SIZE._1)
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: Shouldn't the method fall back to  `"spark.ext.h2o.external.cluster.num.h2o.nodes"` if `"spark.ext.h2o.external.cluster.size"` is not set?

Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: Yup, did that in train, but pushed now :)

**Comment by jakubhava** [comment]
@mn-mikke I'm getting `enable macro paradise to expand macro annotations` failure in tests, wanted to ask if you have seen this when you were working on deprecation utilities. Thanks!

**Comment by mn-mikke** [comment]
@jakubhava As far I remember, i had to add this [`dependency`](https://github.com/h2oai/sparkling-water/pull/1131/files#diff-1d4bec2ed0eb20eda669e2f83c2341c8R16) to get rid of the exception. After that, I haven't see it anymore.

**Comment by jakubhava** [comment]
Thanks!

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala (hunk 2 above)
Comment (mn-mikke) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: If this function is used only explicitly by an user or the original option is very unlikely to be specified then I think we don't have to stress about that. But if other part of SW utilize this function and a customer has the original option stored somewhere in some config (e.g. `spark-defaults.conf`), an upgrade to the next minor version of SW may cause unexpected problems.

Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: Good point, but I think we are still fine as we set the new value based on the deprecated value. I changed the code to centralize the logic. Now if you have the old option specified, and try to use H2OConf, you get warning
```
WARN H2OConf: Please use 'spark.ext.h2o.external.cluster.size' as 'spark.ext.h2o.external.cluster.num.h2o.nodes' is deprecated. Passing the value '1' to 'spark.ext.h2o.external.cluster.size'.
```

**Comment by jakubhava** [comment]
@mn-mikke updated the code, mainly the deprecation so it's more explicit. Can you please check?

H2OConf is either created by the user explicitly or internally in the H2OContext if not specified. Either way, after first contact with it, we check deprecated values and pass them to new corresponding ones.

File: core/src/main/scala/org/apache/spark/h2o/H2OConf.scala [hunk 3]
```diff
@@ -107,7 +110,29 @@ class H2OConf(val sparkConf: SparkConf) extends Logging with InternalBackendConf
...
+        val deprecatedValue = sparkConf.getOption(deprecated)
+        if (deprecatedValue.isDefined) {
+          val currentValue = sparkConf.getOption(current)
+          if (currentValue.isDefined) {
+            logWarning(s"Both options '$deprecated' and '$current' are specified. " +
+              s"Using value '${currentValue.get}' of '$current' as the later one is deprecated.")
+          } else {
+            logWarning(s"Please use '$current' as '$deprecated' is deprecated. Passing the value '${deprecatedValue.get}' to '$current'.")
+          }
+          sparkConf.set(current, deprecatedValue.get)
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/H2OConf.scala]: If both are specified, doesn't you override the current with a value from deprecated?

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala (hunk 2 above)
Comment (mn-mikke) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: Sounds good to me :-)

File: core/src/main/scala/org/apache/spark/h2o/H2OConf.scala [hunk 4]
```diff
@@ -107,7 +110,29 @@ class H2OConf(val sparkConf: SparkConf) extends Logging with InternalBackendConf
   }
 }
 
-object H2OConf {
+object H2OConf extends Logging {
+
+  private val deprecatedOptions = Map("spark.ext.h2o.external.cluster.num.h2o.nodes" -> ExternalBackendConf.PROP_EXTERNAL_CLUSTER_SIZE._1)
+
+  private def checkDeprecatedOptions(sparkConf: SparkConf): Unit = {
+
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/H2OConf.scala]: nit: an extra line?

File: core/src/main/scala/org/apache/spark/h2o/H2OConf.scala (hunk 3 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/H2OConf.scala]: Very good catch!

**Pull request merged**